src/components/GeneratorForm.js

```diff
@@ -0,0 +1,175 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+import SelectInput from "./SelectInput.js";
+import randomFromArray from "./randomFromArray.js";
+
+const GeneratorForm = (props) => {
+    // setState for each input:
+    // NAME INPUT ---
+    const [characterName, setCharacterName] = useState("");
+    // setState for name randomizer:
+    const [randomNameArray, setRandomNameArray] = useState([]);
+    const [randomName, setRandomName] = useState("");
+    // DROPDOWN MENUS ---
+    // character class:
+    const [characterClassArray, setCharacterClassArray] = useState([]);
+    const [characterClass, setCharacterClass] = useState("placeholder");
+    // character race:
+    const [characterRaceArray, setCharacterRaceArray] = useState([]);
+    const [characterRace, setCharacterRace] = useState("placeholder");
+    // character alignment:
+    const [alignmentArray, setAlignmentArray] = useState([]);
+    const [characterAlignment, setCharacterAlignment] = useState("placeholder");
+
+    // establish arrays from API:
+    useEffect(() => {
+        // random character names:
+        axios({
+            url: "https://api.fungenerators.com/name/generate",
+            params: {
+                api_key: "ZMLPKWMnz0eG28IhK5PuLAeF",
+                category: "elf",
+                limit: 200
+            }
+        }).then((returnedName) => {
+            setRandomNameArray(returnedName.data.contents.names)
+            setRandomName(returnedName.data.contents.names[5])
```

Comment:
out of curiosity here, are you always pulling the 5th name returned from your axios call? Would it return the same name every time, or does fungenerators randomize the results?

Reply:
fungenerators SHOULD return random results but sometimes doesn't like to and will return the same list - sometimes it will get stuck on it's first name but will randomize after that, so the safest number was just somewhere (anywhere) higher than 0